app/helpers/application_helper.rb

```diff
@@ -1,2 +1,10 @@
 module ApplicationHelper
+  def link_to_add_fields(name, f, association)
```

Comment:
First of all it will receive 3 parameters, first is the name of the nested field , second is the form and the third is the association name.

``` new_object = f.object.send(association).klass.new```
creates a new instance of associated object

then ```fields_for``` is used to create form for associated object and there is a block which is yielded from fields_for method and it renders the partial view of associated object.

At last it returns a ```link_to``` to call the newly created field